src/Board.cpp

```diff
@@ -0,0 +1,656 @@
+#include "Board.h"
+#include "Rook.h"
+#include "Knight.h"
+#include "Bishop.h"
+#include "Queen.h"
+#include "King.h"
+#include "Pawn.h"
+
+// red color for the pieces of the first player:
+#define RED "\033[31m"
+
+// green color for the pieces of the second player:
+#define GREEN "\033[32m"
+
+// reset color:
+#define RESET "\033[0m"
+
+Board::Board() : board(8, std::vector<Piece*>(8, nullptr)) {
+
```

Comment:
creating the pieces manually and not by the string passed to the chess class
if i change the string for testing i can play with empty pieces